manifest/manifest.go

```diff
@@ -0,0 +1,150 @@
+package manifest
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/ipfs/go-datastore"
+	"golang.org/x/xerrors"
+)
+
+type OnManifestChange func(ctx context.Context, initialInstance uint64, rebootstrap bool, errCh chan error)
+
+type ManifestProvider interface {
+	// Run starts any background tasks required for the operation
+	// of the manifest provider.
+	Run(context.Context, chan error)
+	// Returns the list of gpbft options that should be used
+	GpbftOptions() []gpbft.Option
+	// suscribe to manifest updates
+	Subscribe() <-chan struct{}
+	// Set callback for manifest changes
+	SetManifestChangeCb(OnManifestChange)
+
+	// Manifest accessors
+	MsgPubSubTopic() string
+	NetworkName() gpbft.NetworkName
+	BootstrapEpoch() int64
+	DatastorePrefix() datastore.Key
+	InitialPowerTable() []gpbft.PowerEntry
+	EcConfig() *EcConfig
+}
+
+type Version string
+
+type GpbftConfig struct {
+	Delta                time.Duration
+	DeltaBackOffExponent float64
+	MaxLookaheadRounds   uint64
+}
+
+type EcConfig struct {
+	ECFinality       int64
+	ECDelay          time.Duration
+	CommiteeLookback uint64
+
+	//Temporary
```

Comment:
Can you expand on this please? As in to be removed after passive testing? If so, i would leave an explicit `// TODO: `

Reply:
Chiming @Kubuxu in again. This is not temporary from the passive testing point of view, but I guess the `FakeEC`  one which uses strict timing assumptions.

Reply:
I think we should try removing them; they should only be used as part of the FakeEC, which is outliving its purpose. If we have to hardcode them or make the standalone node inoperable, I'm fine with that (as long as we have tests that Alfonso is adding).

I'm fine with us breaking the standalone node for cleanup purposes, especially if it is FakeEC as it will need a lightweight FilecoinAPI client in there.